packages/playback-core/src/index.ts

```diff
@@ -230,12 +228,66 @@ export const generatePlayerInitTime = () => {
 
 export const generateUUID = mux.utils.generateUUID;
 
-export const toMuxVideoURL = (playbackId?: string, { domain = MUX_VIDEO_DOMAIN, maxResolution = '' } = {}) => {
-  if (!playbackId) return undefined;
-  const [idPart, queryPart = ''] = toPlaybackIdParts(playbackId);
-  const url = new URL(`https://stream.${domain}/${idPart}.m3u8${queryPart}`);
-  if (maxResolution) {
-    url.searchParams.set('max_resolution', maxResolution);
+type MuxVideoURLProps = Partial<{
+  playbackId: string;
+  customDomain: string;
+  maxResolution: MaxResolutionValue;
+  minResolution: MinResolutionValue;
+  renditionOrder: RenditionOrderValue;
+  tokens: Partial<{
+    playback: string;
+    storyboard: string;
+    thumbnail: string;
+  }>;
+  extraPlaylistParams: Record<string, any>;
+}>;
+
+export const toMuxVideoURL = ({
```

Comment:
NOTE: Updated the params here so we can directly pass in react props or the web component instance, hopefully improving maintainability, particularly across components/elements.